### DISCUSSION TITLE
Lavalink for arm linux ubuntu server

### DISCUSSION THREAD

**Discussion by MrKsiJ** [lavalink-devs/Lavalink#863] [Q&A] [answered]
please throw here a link to the latest release of Lavalink that supports OpenJDK 1.8.0 I use a supported version of Java-8-10 on my arm linux server, but the bot does not want to play music in the head channel on this version, everything works on the latest version 3.7.5, but it is not suitable for my arm linux server because openjdk No one has released 11 versions for arm processors yet (version 3.7.5 was tested on VMware)

**Comment by DRSchlaubi** [comment]
There are Java 11 binaries for arm? https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.18%2B10

Lavalink uses Java 11 for quite some time, even if there is a version using Java 8, it probably will no longer work, due to the new voice ip discovery packet

**Reply by MrKsiJ** [reply]
if version 3.2.2 does not take off on the server, I will try to unpack binaries openjdk 11 for arm, if it works, it will be cool thanks)

**Reply by DRSchlaubi** [reply]
Just use the package manager https://adoptium.net/de/installation/linux/#_deb_installation_on_debian_or_ubuntu

**Reply by MrKsiJ** [reply]
> There are Java 11 binaries for arm? https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.18%2B10
> 
> Lavalink uses Java 11 for quite some time, even if there is a version using Java 8, it probably will no longer work, due to the new voice ip discovery packet

the binaries on my armhf processor work is cool

**Comment by topi314** [comment]
I'm pretty sure you can just install java 11 with
```sh
sudo apt update
sudo apt install default-jdk
```
on a raspberry pi
just run
```sh
java -version
```
to confirm the version

**Reply by MrKsiJ** [reply]
default-jdk automatically installs version 1.8.0 for my arm server on the armhf platform

**Reply by topi314** [reply]
you might need to update your pi version. 
Also heavily depends on which pi you have, so hard to tell

**Reply by MrKsiJ** [reply]
> you might need to update your pi version. Also heavily depends on which pi you have, so hard to tell

on the developer's website, the latest version of orangepi 3.1.4 5 version does not work on my orange pi

**Reply by topi314** [reply]
yeah I can't help you with orangepi. I only have experience with raspberry pis

**Reply by aikaterna** [reply]
Azul is the only provider I know of that offers armhf Java 11 builds. https://www.azul.com/downloads

Filters will not work on an armhf processor, so if you're intending to use any transforms like timescale you will be out of luck/can possibly segfault your Lavalink.jar.

If you can, try acquiring a PI 3b+ or similar device minimum, as running a bot + Lavalink on an armhf device sounds rather painful.

Additional info edit: Lavalink can't run on Java 8 because some of the dependencies have a minimum of Java 10, like koe I believe.

**Comment by MrKsiJ** [comment, marked as answer]
I was able to install 11 java openjdk I didn't think it would start
https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.18%2B10/OpenJDK11U-jdk_arm_linux_hotspot_11.0.18_10.tar.gz
